src/hooks/useComponentVisible.ts

```diff
@@ -0,0 +1,41 @@
+import { useState, useRef, useEffect } from 'react'
+
+const useComponentVisible = () => {
+  const [isComponentVisible, setIsComponentVisible] = useState(false)
+  const componentRef = useRef<any>(null)
+  const triggerRef = useRef<HTMLButtonElement>(null)
```

Comment:
It's better to be more specific - the trigger should ideally be a button so there's nothing wrong with specifying that here. If we need it changed in the future it's a small enough change then. As for `componentRef` on line 5, that got complicated with TypeScript since the `ListEditForm` component demanded a particular ref type and this required something different. So we changed both to use `RefObject<any>` or `LegacyRef<any>` types instead of specifying a particular HTML element.